src/openapi/streaming/connection/transport/signalr-core-transport.js

```diff
@@ -334,58 +330,84 @@ SignalrCoreTransport.prototype.updateQuery = function(
     });
 
     this.contextId = contextId;
-    this.authToken = authToken;
+    this.authToken = authToken.replace('BEARER ', '');
 
     if (forceAuth) {
-        this.renewSession(authToken, contextId);
+        this.renewSession(this.authToken, contextId);
     }
 };
 
 SignalrCoreTransport.prototype.renewSession = function(authToken, contextId) {
-    const options = {
-        headers: {
-            Authorization: authToken,
-        },
-    };
-    const url = `${this.baseUrl}/streaming/renewal/renewsession`;
+    if (!this.connection || this.isReconnecting) {
+        return;
+    }
 
-    return fetch('POST', url, options)
-        .then(() => {
-            log.debug(
-                LOG_AREA,
-                'Streaming session is successfully renewed with new token',
-                {
-                    contextId,
-                },
-            );
+    return this.connection
+        .invoke('RenewToken', authToken)
+        .then(({ Status }) => {
+            switch (Status) {
+                case renewStatus.SUCCESS:
+                    log.info(
+                        LOG_AREA,
+                        'Successfully renewed token for session',
+                        {
+                            contextId,
+                        },
+                    );
+                    return;
+
+                case renewStatus.INVALID_TOKEN:
+                    // if this call was superseded by another one, then ignore this error
+                    // else let auth provider know of invalid token
+                    if (this.authToken === authToken) {
+                        this.unauthorizedCallback();
+                    }
+
+                    return;
+
+                case renewStatus.SESSION_NOT_FOUND:
+                    log.warn(
+                        LOG_AREA,
+                        'Session not found while renewing session',
+                        {
+                            contextId,
+                        },
+                    );
+
+                    // should try to reconnect with different context id
+                    this.stop();

```

Comment:
should we stop? what if then then gets called back but we are already reconnecting or have reconnected?

Reply:
I don't think that's even possible since we are using websocket messages.
if we are in reconnecting state that means this callback will get cancelled before triggering  reconnecting state
and for 2nd case where we are trying to renew when we are reconnecting, thats already being handled